src/main/scala/com/advancedtelematic/director/db/DeviceUpdate.scala

```diff
@@ -12,10 +12,12 @@ import slick.jdbc.MySQLProfile.api._
 object DeviceUpdateResult {
   sealed abstract class DeviceUpdateResult
 
-  final case object NoChange extends DeviceUpdateResult
-  final case class UpdatedSuccessfully(updateTarget: DeviceUpdateTarget) extends DeviceUpdateResult
-  // the Option on targets is if the device have targets or not
-  final case class UpdatedToWrongTarget(timestamp: Int, targets: Option[Map[EcuSerial, Image]], manifest: Map[EcuSerial, Image]) extends DeviceUpdateResult
+  final case object NoUpdate extends DeviceUpdateResult
+  final case class UpdateNotCompleted(updateTarget: DeviceUpdateTarget) extends DeviceUpdateResult
+  final case class UpdateSuccessful(updateTarget: DeviceUpdateTarget) extends DeviceUpdateResult
+  final case class UpdateUnexpectedTarget(
+    updateTarget: DeviceUpdateTarget, targets: Map[EcuSerial, Image], manifest: Map[EcuSerial, Image]
```

Comment:
Since `targets` and `manifest` share the same type and, if I understand this correctly, one represents the expected targets and the other one the actual targets, maybe naming them something like `actualTargets` and `expectedTargets` would make that more evident?